File: content-solutions/backtracking/outfit-picker.js
function solution(closet) {
  const result = [];
  const path = [];
  function choose(i) {
    if (i === closet.length) {
      result.push([...path]);
      return;
    }
    for (const item of closet[i]) {
      path.push(item);
      choose(i + 1);
      path.pop();
    }
  }
  choose(0);
  return result;
}

File: content-solutions/backtracking/outfit-picker.py
def solution(closet):
    result = []
    path = []

    def choose(i):
        if i == len(closet):
            result.append(path[:])
            return
        for item in closet[i]:
            path.append(item)
            choose(i + 1)
            path.pop()

    choose(0)
    return result
